packages/lesswrong/components/messaging/NewConversationDialog.tsx

```diff
@@ -0,0 +1,196 @@
+      display: "inline-block",
+      position: "relative",
+      width: "100%",
+      marginLeft: 12,
+      height: 46,
+      whiteSpace: "nowrap",
+      boxSizing: "border-box",
+    },
+    "& .ais-SearchBox-form": {
+      height: "100%",
+    },
+    "& .ais-SearchBox-submit": {
+      display: "none",
+    },
+    // This is a class generated by React InstantSearch, which we don't have direct control over so
+    // are doing a somewhat hacky thing to style it.
+    "& .ais-SearchBox-input": {
+      height: "100%",
+      width: "100%",
+      paddingRight: 0,
+      verticalAlign: "bottom",
+      borderStyle: "none",
+      boxShadow: "none",
+      backgroundColor: "transparent",
+      fontSize: "inherit",
+      "-webkit-appearance": "none",
+      cursor: "text",
+      ...theme.typography.body2,
+    },
+  },
+  searchHelp: {
+    [theme.breakpoints.down("sm")]: {
+      display: "none",
+    },
+  },
+  infoIcon: {
+    fontSize: 20,
+    fill: theme.palette.grey[800],
+  },
+  usersList: {
+    overflowY: "auto",
+    // Not the actual height, but makes it fill the space when there are no results
+    height: 1000,
+  },
+  hit: {
+    paddingLeft: 20,
+    paddingRight: 20,
+  },
+  closeIcon: {
+    color: theme.palette.grey[600],
+    width: 20,
+    height: 20,
+    cursor: "pointer",
+  }
+});
+
+const NewConversationDialog = ({
+  isModInbox = false,
+  classes,
+  onClose,
+}: {
+  isModInbox?: boolean;
+  classes: ClassesType;
+  onClose: () => void;
+}) => {
+  const { LWDialog, ErrorBoundary, ExpandedUsersConversationSearchHit, ForumIcon, LWTooltip, Typography } = Components;
+  const currentUser = useCurrentUser();
+  const [query, setQuery] = useState<string>("");
+
+  if (!currentUser) return null;
+
+  return (
+    <AnalyticsContext pageSectionContext="newConversationDialog">
+      <LWDialog open={true} onClose={onClose}>
+        <div className={classes.root}>
+          <div className={classes.titleRow}>
+            <div>New conversation</div>
+            <ForumIcon icon="Close" className={classes.closeIcon} onClick={onClose} />
+          </div>
+          <InstantSearch
+            indexName={getElasticIndexNameWithSorting("Users", "relevance")}
+            searchClient={getSearchClient()}
+            searchState={{ query }}
+            onSearchStateChange={(x) => setQuery(x.query)}
+          >
+            <div className={classes.resultsColumn}>
+              <div className={classes.searchBoxRow}>
+                <div className={classes.searchInputArea}>
+                  <ForumIcon icon="Search" className={classes.searchIcon} />
+                  {/* Ignored because SearchBox is incorrectly annotated as not taking null for its reset prop, when
+                    * null is the only option that actually suppresses the extra X button.
+                  // @ts-ignore */}
+                  <SearchBox defaultRefinement={query} reset={null} focusShortcuts={[]} autoFocus={true} />
```

Comment:
The designs include improvements to this search box, I've punted those to a followup ticket: https://app.asana.com/0/628521446211730/1205817215822667/f